contracts/governancestaking/BaseGovernanceShareField.sol

```diff
@@ -0,0 +1,203 @@
+				mintCumulation = mintCumulation + pending;
+			}
+		} 
+	
+
+	
+
+	/**
+	 * @dev This function increase user's productivity and updates the global productivity.
+	 * This function increase user's productivity and updates the global productivity.
+	 * the users' actual share percentage will calculated by:
+	 * Formula:     user_productivity / global_productivity
+	 */
+	function _increaseProductivity(address user, uint256 value)
+		internal
+		virtual
+		returns (bool)
+	{
+		require(value > 0, "PRODUCTIVITY_VALUE_MUST_BE_GREATER_THAN_ZERO");
+
+		UserInfo storage userInfo = users[user];
+		_update();
+		_audit(user);
+
+		totalProductivity = totalProductivity + value;
+		userInfo.amount = userInfo.amount + value;
+		userInfo.rewardDebt = (userInfo.amount * accAmountPerShare) / 1e2; // Divide to 1e2 to keep rewardDebt in 2 decimals
+		return true;
+	}
+
+	/**
+	 * @dev This function will decreases user's productivity by value, and updates the global productivity
+	 * it will record which block this is happenning and accumulates the area of (productivity * time)
+	 */
+
+	function _decreaseProductivity(address user, uint256 value)
+		internal
+		virtual
+		returns (bool)
+	{
+		UserInfo storage userInfo = users[user];
+		require(
+			value > 0 && userInfo.amount >= value,
+			"INSUFFICIENT_PRODUCTIVITY"
+		);
+
+		_update();
+		_audit(user);
+		
+		userInfo.amount = userInfo.amount - value;
+		userInfo.rewardDebt = (userInfo.amount * accAmountPerShare) / 1e2;
+		totalProductivity = totalProductivity - value;
+
+		return true;
+	}
+
+	/**
+	 * @dev Query user's pending reward with updated variables
+	 * @return returns  amount of user's earned but not minted rewards
+	 */
+	function getUserPendingReward(address user) public view returns (uint256) {
+		UserInfo memory userInfo = users[user];
+		uint256 _accAmountPerShare = accAmountPerShare;
+		// uint256 lpSupply = totalProductivity;
+	
+		uint256 pending = 0;
+	
+		if (totalProductivity != 0) 
+        {
+			uint256 multiplier = block.number - lastRewardBlock;
+			uint256 reward = multiplier * rewardsPerBlock; // rewardsPerBlock is in GDAO which is in 0 decimals
+		
+
+			_accAmountPerShare = _accAmountPerShare + rdiv(reward * 1e2 ,totalProductivity) / 1e25; //multiply reward by 1e2 so bring it to 2 decimals as well then divide result by 1e25 so accAmountPerShare is in 2 decimals
+			
+			
+			pending =
+					(userInfo.amount * accAmountPerShare) / 1e2
+					- userInfo.rewardDebt; // Divide 1e2 to reduce 2 Decimals since rewardDebt in 2 decimals so we can calculate how much reward earned in that cycle
+				
+			
+		}
+		return userInfo.rewardEarn + pending / 1e2; // Reward earn in 2decimals so need to divide 1e2 to bring down GDAO decimals which is 0
+	}
+
+	/** 
+    @dev Mint earned rewards of the user and update their reward info
+    * @param user address of the user that will be accounted
+    * @return returns minted amount
+    */
+
+	function _mintRewards(address user)
```

Comment:
the minting logic should be in inherting contract
this contract should only be about calculations

Reply:
changed